src/Storage/Imap.php

```diff
@@ -354,16 +354,31 @@ public function getFolders($rootFolder = null)
      * @throws Exception\RuntimeException
      * @throws \Zend\Mail\Protocol\Exception\RuntimeException
      */
-    public function selectFolder($globalName)
+    public function selectFolder($globalName, $examine = false)
     {
         $this->currentFolder = $globalName;
-        if (!$this->protocol->select($this->currentFolder)) {
+        $method = ($examine ? 'examine' : 'select');
+        if (!$this->protocol->$method($this->currentFolder)) {
             $this->currentFolder = '';
             throw new Exception\RuntimeException('cannot change folder, maybe it does not exist');
         }
     }
 
     /**
+     * examine given folder
```

Comment:
There's nothing better than quoting the RFC:

```
The EXAMINE command is identical to SELECT and returns the same
      output; however, the selected mailbox is identified as read-only.
      No changes to the permanent state of the mailbox, including
      per-user state, are permitted; in particular, EXAMINE MUST NOT
      cause messages to lose the \Recent flag.
```

Anyone who uses it should know what it is, just as they know when to use `SELECT`.